src/main/java/Duke/Duke.java

```diff
@@ -0,0 +1,211 @@
+            dateStartIndex = itemEndIndex + 5;
+            try {
+                desc = command.substring(itemStartIndex, itemEndIndex);
+                date = command.substring(dateStartIndex);
+            }catch(StringIndexOutOfBoundsException e) { // if task or by date was not entered, catch error
+                out = "Invalid deadline task format!"
+                    + " Try the following: deadline [task] /by [date] ";
+                break;
+            }
+            out = list.addTask(desc, listTypes.deadline, date);
+            break;
+        case "event": // If command is event, parse the task and at date, print confirmation
+            itemEndIndex = command.indexOf("/at") - 1;
+            dateStartIndex = itemEndIndex + 5;
+            try { // if task or at date was not entered, catch error
+                desc = command.substring(itemStartIndex, itemEndIndex);
+                date = command.substring(dateStartIndex);
+            }catch(StringIndexOutOfBoundsException e) {
+                out = "Invalid command format!"
+                    + " Try the following: event [task] /at [date] ";
+                break;
+            }
+            out = list.addTask(desc, listTypes.event, date);
+            break;
+        case "todo": // If command is todo, parse the task, print confirmation
+            desc = command.substring(itemStartIndex);
+            if(desc.equals("todo")){ // if task was not entered, catch error
+                out = "Invalid todo task!";
+                break;
+            }
+            out = list.addTask(desc,listTypes.todo);
+            break;
+        case "done": // If command is done, mark task as done, print confirmation
+            try {
+                out = list.resolveTask(tokens[1]);
+            }catch(ArrayIndexOutOfBoundsException e){
+                out = "Invalid selection!";
+                break;
+            }
+            break;
+        case "delete": // If command is  delete, delete task entry in list, print confirmation
+            try {
+                out = list.deleteTask(tokens[1]);
+            }catch(IndexOutOfBoundsException e){ // If task number is invalid, catch error
+                out = "Invalid selection!";
+                break;
+            }catch(NumberFormatException e){ // If string is entered instead of task number, catch error
+                out = "Please enter the number of the task!";
+            }
+            break;
+
+        default: // If command entered is any other unrecognised command, print invalid command
+            out = "Invalid task command!";
+        }
+        return out;
+    }
+
+
+    public static void importParsing(todoList list, String input){ // Parses the inputted line and executes the respective commands to import the tasks through command parser
+        String[] tokens = input.split("/");
+        String command = "";
+        switch(tokens[0]){
+        case "D":
```

Comment:
Single characters as token names might not be the best way to describe what the case is about. See if you could try to increase meaning, as described [here](https://nus-cs2113-ay2021s2.github.io/website/se-book-adapted/chapters/codeQuality.html#not-too-long-not-too-short).